plot/plot.app.js

```diff
@@ -684,13 +690,57 @@
 
                                                     data.forEach(function (row) {
                                                         if(label) {
-                                                            values.labels.push(row[trace.legend_col]);
+                                                            var traceLabel=row[trace.legend_col];
+                                                            //Changes related to adding markdown templating pattern to legends in pie
+                                                            if(trace.hasOwnProperty("legend_markdown_pattern") && trace.legend_markdown_pattern){
+                                                                // Check if the ? is present in the markdown pattern if yes add & else add ?
+                                                                var qCharacter = /\((.*?)\)/ig.exec(trace.legend_markdown_pattern)[1] !== -1 ? "&" : "?";
+                                                                $rootScope.templateParams = {
+                                                                    $traces: {
+                                                                        data: data,
+                                                                    },
+                                                                    $self:{
+                                                                        data: row
+                                                                    }
+                                                                }
+                                                                // $rootScope.templateParams= {
+                                                                //     Schema_Table: row['Schema_Table'],
+                                                                //     Data_Type_Filter: ERMrest.encodeFacet(row['Data_Type_Filter']),
+                                                                //     Title: traceLabel,
+                                                                // };
+                                                                try{
```

Comment:
I have put in the try and catch block as if the traceLabel is null it will break the regex and subsequent lines of code.. In catch, if the link fails due to some reason, do we want to throw an error or do we want to have no link for that label?